app/main.py

```diff
@@ -1 +1,51 @@
-#  write your code here
+class Animal:
+
+    def __init__(
+            self, name: str,
+            appetite: int = 0,
+            is_hungry: bool = True
+
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(
+            self, name: str,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(name=name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(
+            self, name: str,
+            is_hungry: bool = True,
+    ) -> None:
+        super().__init__(name=name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    total_food_points = sum(
+        [animal.feed() for animal in animals if animal.is_hungry]
+    )
```

Comment:
You can make this more efficient by using a generator expression. Generator expressions are similar to list comprehensions, but they don't create the whole list at once. Instead, they generate each value on the fly. You can achieve this simply by removing the square brackets